heron/tools/apiserver/src/java/com/twitter/heron/apiserver/resources/FileResource.java

```diff
@@ -143,4 +153,12 @@ private Config createConfig() {
     return Config.toLocalMode(builder.build());
   }
 
+  private String getHostNameOrIP() {
+    // Override hostname if provided in flags
+    if (StringUtil.isNotBlank(getDownloadHostName())) {
+      hostname = getDownloadHostName();
+    }
+    return (hostname != null) ? hostname : ip.toString();
+  }
+
 }
```

Comment:
I think we should not be setting static state here.
final String downloadHostName = getDownloadHostName();
if (StringUtil.isNotBlank(downloadHostName) {
  return downloadHostName;
}

return hostname != null ? hostname : ip;

Reply:
Changed it and added a check for ip being null